doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+a math context because it does not render as the Greek letter (α).
+
+Style Preferences
+-----------------
+
+**Spelling and Punctuation**
+
+All narrative writing in SymPy follows American spelling and punctuation
+standards. For example, “color” is preferred over “colour” and commas should be
+placed inside of quotation marks.
+
+**Examples**
+
+::
+
+  If the ``line_color`` aesthetic is a function of arity 1, then the coloring is
+  a function of the x value of a point.
+
+  The term "unrestricted necklace," or "bracelet," is used to imply an object
+  that can be turned over or a sequence that can be reversed.
+
+If there is any ambiguity about the spelling of a word, such as in the case of
+a function named after a person, refer to the spelling of the actual SymPy
+function.
+
+For example, Chebyshev polynomials are named after Pafnuty Lvovich Tchebychev,
+whose name is sometimes transliterated from Russian to be spelled with a “T,”
+but in SymPy it should always be spelled “Chebyshev” to refer to the SymPy
+function.
+
+**Example**
+
+::
+
+  class chebyshevt(OrthogonalPolynomial):
+      r"""
+      Chebyshev polynomial of the first kind, $T_n(x)$
+      ...
+
+      """
+
+**Capitalization**
+
+Title case capitalization is preferred in all SymPy headings.
+
+**Example**
+
+::
+
+  What is Symbolic Computation?
+  -----------------------------
+
+**Grammar**
+
+WORK IN PROGRESS: Google Season of Docs technical writer will track common
+grammar mistakes (e.g., using certain articles correctly) while editing and add
+to this guide.
+
+Tone Preferences
+----------------
+
+Across SymPy documentation, please write in:
+
+* The present tense (e.g., In the following section, we are going to learn...)
+* The first-person inclusive plural (e.g., We did this the long way, but now we
+  can try it the short way...)
+* Use the generic pronoun “you” instead of “one.” Or use “the reader” or “the
+  user.” (e.g., You can access this function by... The user can then access
+  this function by...)
+* Use the gender-neutral pronoun “they” instead of “he” or “she.” (e.g., A good
+  docstring tells the user exactly what they need to know.)
+
+Avoid extraneous or belittling words such as “obviously,” “easily,” “simply,”
+“just,” or “straightforward.”
+
+Avoid unwelcoming or judgement-based phrases like “That is wrong.” Instead use
+friendly and inclusive language like “A common mistake is...”
+
+Avoid extraneous phrases like, “we just have to do one more thing.”
+
+Docstring Guidelines
+====================
+
+To contribute to SymPy’s docstrings, please read these guidelines in full.
+
+A documentation string (docstring) is a string literal that occurs as the first
+statement in a module, function, class, or method definition. Such a docstring
+becomes the ``__doc__`` special attribute of that object.
+
+**Example**
+
+Here is a basic docstring::
+
+  def fdiff(self, argindex=1):
+      """
+      Returns the first derivative of a Heaviside Function.
```

Comment:
I think there should be more explanation on how to write the one-line summary. I personally dislike using "Returns" as a summary of what a function/method does. Normally I would write this one line summary not so much as a sentence but as either a noun like
```
First derivative of the Heaviside function.
```
or as an imperative (if the function is non-pure) like
```
Set the ith element of the matrix.
```